src/main/java/ch/njol/skript/expressions/ExprWhitelist.java

```diff
@@ -38,83 +40,112 @@
 import ch.njol.util.coll.CollectionUtils;
 
 @Name("Whitelist")
-@Description("A server's whitelist." +
-	"This expression can be used to add/remove players to/from the whitelist," +
-	" to enable it and disable it (set whitelist to true / set whitelist to false)," +
-	" and to empty it (reset whitelist)")
+@Description({
+	"A server's whitelist or whitelist enforcement.",
+	"This expression can be used to add/remove players to/from the whitelist.",
+	"To enable and disable it (set whitelist to true / set whitelist to false), and to empty it (reset whitelist)",
+	"To enable and disable enforcement (set whitelist enforcement to true / set whitelist enforcement to false),",
+	"which kicks all non-whitelisted players once set to true."
+})
 @Examples({"set whitelist to false",
 	"add all players to whitelist",
 	"reset the whitelist"})
-@Since("2.5.2")
+@Since("2.5.2, INSERT VERSION (enforce)")
+@RequiredPlugins("Minecraft 1.17+")
 public class ExprWhitelist extends SimpleExpression<OfflinePlayer> {
-	
+
+	private static final boolean ENFORCE_SUPPORT;
+
 	static {
-		Skript.registerExpression(ExprWhitelist.class, OfflinePlayer.class, ExpressionType.SIMPLE, "[the] white[ ]list");
+		ENFORCE_SUPPORT = Skript.methodExists(Bukkit.class, "setWhitelistEnforced", boolean.class);
```

Comment:
Maybe enforcement should be a separate effect?
Just `[un]enforce the whitelist`